app/platform/fabric/sync/SyncService.ts

```diff
@@ -375,25 +383,63 @@ export class SyncServices {
 		this.synchInProcess.push(synch_key);
 
 		const channel_genesis_hash = client.getChannelGenHash(channel_name);
-		const blockHeight = parseInt(channelInfo.height.low) - 1;
-		// Query missing blocks from DB
-		const results = await this.persistence
+		const latestBlockHeight = parseInt(channelInfo.height.low) - 1;
+
+		// Get the value of noOfBlocks from configuration file
+		const noOfBlocks = network_configs[network_id].noOfBlocks;
+
+		// Calculate the starting block height for sync
+		const startingBlockHeight = Math.max(0, latestBlockHeight - noOfBlocks + 1);
+
+		// Syncing Details
+		logger.info(
+			`Syncing blocks from ${startingBlockHeight} to ${latestBlockHeight}`
+		);
+
+		// Load the latest blocks as per the configuration
+		for (
+			let blockHeight = latestBlockHeight;
+			blockHeight >= startingBlockHeight;
+			blockHeight--
+		) {
+			try {
+				const block = await client.fabricGateway.queryBlockByNumber(
+					channel_name,
+					blockHeight
+				);
+				if (block) {
+					await this.processBlockEvent(client, block, noDiscovery);
+				}
+				logger.info(`Synced block #${blockHeight}`);
+			} catch {
+				logger.error(`Failed to process Block # ${blockHeight}`);
+			}
+		}
+
+		const missingBlocks = await this.persistence
```

Comment:
Can you check whether data is flowing through the missingBlocks section. since when we have new blocks added, latest blocks are also synced and added into db in the above section.

Reply:
Yes, data is flowing through the missingBlocks section as expected. As new blocks are added, the latest blocks are being synced and added into the database in the above section without any issues. Everything seems to be working as expected.